content/blog/2022/2022-08-30-handling-timezones-in-spring.md

```diff
@@ -0,0 +1,516 @@
+
+Sample conversion example:
+
+````java
+        OffsetDateTime currentDateTime = OffsetDateTime.now();
+        System.out.println("System default timezone current zone offset date/time : " + currentDateTime);
+
+        // Check for format difference
+        ZonedDateTime currentZoneDateTime = ZonedDateTime.now();
+        System.out.println("Current system zone date/time : " + currentZoneDateTime);
+
+        ZoneOffset zoneOffSet= ZoneOffset.of("+01:00");
+        OffsetDateTime offsetDateTime = OffsetDateTime.now(zoneOffSet);
+        System.out.println("Europe/London zone offset date/time : " + offsetDateTime);
+
+        OffsetDateTime fromLocals = OffsetDateTime.of(LocalDate.now(), LocalTime.now(), currentDateTime.getOffset());
+        System.out.println("Get Offset date/time from Locals : " + fromLocals);
+
+        OffsetDateTime fromLocalDateTime = OffsetDateTime.of(LocalDateTime.of(2022, Month.NOVEMBER, 1, 10, 10, 10), currentDateTime.getOffset());
+        System.out.println("Get Offset date/time from LocalDateTime with Offset at the current Instant considered " +
+                "(does not consider DST at custom date): " + fromLocalDateTime);
+
+        OffsetDateTime fromLocalsWithDefinedOffset = OffsetDateTime.of(LocalDate.now(), LocalTime.now(), ZoneOffset.systemDefault().getRules().getOffset(LocalDateTime.of(2022, Month.NOVEMBER, 1, 10, 10, 10)));
+        System.out.println("Get Offset date/time from Local with offset for custom LocalDateTime considered " +
+                "(Considers DST at custom date) : " + fromLocalsWithDefinedOffset);
+
+        OffsetDateTime sameInstantDiffOffset = currentDateTime.withOffsetSameInstant(ZoneOffset.of("+01:00"));
+        System.out.println("Same instant at a different offset : " + sameInstantDiffOffset);
+
+        OffsetDateTime dt = OffsetDateTime.parse("2011-12-03T10:15:30+01:00", DateTimeFormatter.ISO_OFFSET_DATE_TIME);
+        DateTimeFormatter fmt = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss'Z'");
+        System.out.println("OffsetDateTime parsed and formatted" + fmt.format(dt));
+
+        OffsetDateTime convertFromZoneToOffset = currentZoneDateTime.toOffsetDateTime();
+        System.out.println("Convert from ZonedDateTime to OffsetDateTime : " + convertFromZoneToOffset);
+
+````
+
+Output:
+
+{{% image alt="settings" src="images/posts/handling-timezones-in-spring/offsetDateTimeOutput.JPG" %}}
+
+## Compatibility with the legacy API
+
+As a part of the Date/Time API, **methods have been introduced to convert from legacy classes to the newer API objects**.
+
+Sample Code Conversion:
+
+````java
+
+    @Test
+    public void testWorkingWithLegacyDateInJava8() {
+        Date date = new Date();
+        System.out.println("java.util.Date : " + date);
+        Instant instant = date.toInstant();
+        System.out.println("Convert java.util.Date to Instant : " + instant);
```

Comment:
Please use assertions instead of `System.out.println()`, because the reader would have to guess what comes out of a `System.out.println()` (or scroll down to an "output" section)